tests/Tests/ORM/Mapping/ReflectionReadonlyPropertyTest.php

```diff
@@ -17,36 +20,83 @@
  */
 class ReflectionReadonlyPropertyTest extends TestCase
 {
-    public function testSecondWriteWithSameValue(): void
+    /**
+     * @dataProvider sameValueProvider
+     */
+    public function testSecondWriteWithSameValue(object $entity, string $property, mixed $value, mixed $sameValue): void
     {
-        $author = new Author();
-
-        $wrappedReflection = new ReflectionProperty($author, 'name');
+        $wrappedReflection = new ReflectionProperty($entity, $property);
         $reflection        = new ReflectionReadonlyProperty($wrappedReflection);
 
-        $reflection->setValue($author, 'John Doe');
+        $reflection->setValue($entity, $value);
 
-        self::assertSame('John Doe', $wrappedReflection->getValue($author));
-        self::assertSame('John Doe', $reflection->getValue($author));
+        self::assertSame($value, $wrappedReflection->getValue($entity));
+        self::assertSame($value, $reflection->getValue($entity));
 
-        $reflection->setValue($author, 'John Doe');
+        $reflection->setValue($entity, $sameValue);
 
-        self::assertSame('John Doe', $wrappedReflection->getValue($author));
-        self::assertSame('John Doe', $reflection->getValue($author));
+        /*
+         * Intentionally testing against the initial $value rather than the $sameValue that we just set above one in
+         * order to catch false positives when dealing with object types
+         */
+        self::assertSame($value, $wrappedReflection->getValue($entity));
+        self::assertSame($value, $reflection->getValue($entity));
     }
 
-    public function testSecondWriteWithDifferentValue(): void
+    public function sameValueProvider(): Generator
```

Comment:
Please add a docblock documenting the return type (it should match the signature of `testSecondWriteWithSameValue()`